packages/fxa-event-broker/lib/selfUpdatingService/index.ts

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This service should probably be in fxa-shared as I could see the pattern being useful elsewhere. There's a different version of this I've seen in some Python caching libraries that implements some logic that only does the fetch on-demand (vs background interval like here), which prevents concurrent fetches and anyone after the refresh trigger gets the prior cached value. An abstraction for that pattern would also be handy to have in fxa-shared.